### DISCUSSION TITLE
TDSZ2 Stlink/speed sensor wiring

### DISCUSSION THREAD

**Discussion by Grayhairedbloke** [danielnilsson9/bbs-fw#151] [Q&A]
I have purchase St link V2 emulator and a TDSZ2 speed sensor extension cable in preparation to flash the firmware on m,t TDSZ2 36v motor
In preparing the cable for soldering the the wires I found that the wire colours of the speed sensor do not correspond to the wire colours shown in the wiki on this site.
Image 1 shows the the location of where the colours are in relationship to the pins
![image](https://github.com/danielnilsson9/bbs-fw/assets/81550472/4ffdfa6b-414a-42f3-b90d-431cb216bb8e)
Image 2 shows the actual plug  wiring colours
![image](https://github.com/danielnilsson9/bbs-fw/assets/81550472/33fa0d24-c6bd-4eab-b563-0466eee4ee8b)
Image 3 show my proposed wiring from the female plug to the st link
![image](https://github.com/danielnilsson9/bbs-fw/assets/81550472/503dc8f3-14ff-465d-98a2-6efb760cded5)
 Please advise whether this is where the wires supposed to go.
I need a reply by Wednesday 26 th July as I'm intend to flash the firmware on Friday
Please advise to ubonvet@gmail.com
Thanks so much for your help.

**Comment by danielnilsson9** [comment]
You can never trust the wire colors, they change. You should wire the cable to match how it is done here (positions on connector):
https://github.com/OpenSourceEBike/TSDZ2_wiki/wiki/Flash-the-firmware-on-TSDZ2#user-content-Wire_the_STLinkV2_to_the_TSDZ2_wheel_speed_sensor_cable

TSDZ2 support in this firmware is still experimental. The 36V version on TSDZ2 is NOT currently supported, will try to add it in next version. Also there is a very annoying issue for TSDZ2 in the latest released version of this firmware (1.3.1) making the motor shake produce noise.

I suspect pretty much no one except me and some friends are using the TSDZ2 version of this firmware at the moment.

Also this firmware requires you to use a Bafang UART compatible display, it cannot be used with the displays shipped with TSDZ2, I'm using SW102. You will have to make a custom main cable harness to reroute the signals to a Bafang compatible display. If you have the 8-pin version I have a image somewhere of the pinouts, will post later.
